napari/_vispy/vispy_text_layer.py

```diff
@@ -0,0 +1,152 @@
+        node = Compound([Line(), TextNode(), TextNode()])
+
+        super().__init__(layer, node)
+
+        self.layer.events.rotation.connect(lambda e: self._on_data_change())
+        self.layer.events.font_size.connect(lambda e: self._on_data_change())
+        self.layer.events.text_color.connect(lambda e: self._on_data_change())
+        self.layer.events.highlight.connect(
+            lambda e: self._on_highlight_change()
+        )
+
+        self.layer.dims.events.ndisplay.connect(
+            lambda e: self._on_display_change()
+        )
+
+        self._on_display_change()
+
+    def _on_display_change(self):
+        parent = self.node.parent
+        self.node.transforms = ChainTransform()
+        self.node.parent = None
+
+        if self.layer.dims.ndisplay == 2:
+            self.node = Compound([Line(), TextNode(), TextNode()])
+        else:
+            self.node = Markers()
+
+        self.node.parent = parent
+        self.layer._update_dims()
+        self.layer._set_view_slice()
+        self.reset()
+
+    def _on_data_change(self):
+
+        # Set vispy data, noting that the order of the points needs to be
+        # reversed to make the most recently added point appear on top
+        # and the rows / columns need to be switch for vispys x / y ordering
+        if len(self.layer._text_coords_view) == 0:
+            data = np.zeros((1, self.layer.dims.ndisplay))
+            text = []
+
+        else:
+            data = self.layer._text_coords_view
+            text = self.layer._text_view
+
+        # Update the text
+        if self.layer.dims.ndisplay == 2:
+            positions = np.flip(data, axis=1)
+            text_node = self.node._subvisuals[self._TEXT_NODE_INDEX]
+            self._update_text_node(
+                text_node,
+                text=text,
+                pos=positions,
+                rotation=self.layer.rotation,
+                color=self.layer.text_color,
+                font_size=self.layer.font_size,
+            )
+
+    def _on_highlight_change(self):
+        if self.layer.dims.ndisplay == 3:
+            return
+
+        if len(self.layer._highlight_index) > 0:
+            # Color the hovered or selected points
+            data = self.layer._text_coords_view[self.layer._highlight_index]
+            if data.ndim == 1:
+                data = np.expand_dims(data, axis=0)
+            size = self.layer.font_size
+            text = [
+                self.layer._text_view[i] for i in self.layer._highlight_index
+            ]
+
+        else:
+            data = np.zeros((1, self.layer.dims.ndisplay))
+            size = 0
+            face_color = 'white'
+            text = []
+
+        highlight_text_node = self.node._subvisuals[
+            self._HIGHLIGHT_TEXT_NODE_INDEX
+        ]
+        self._update_text_node(
+            highlight_text_node,
+            text=text,
+            pos=np.flip(data, axis=1),
```

Comment:
this needs a rotation parameter too, right now the highlight is not rotated

Reply:
you might also now have to take rotation into account inside the selection box calculation - it doesn't seem to have been done yet

Reply:
I added the rotation parameter. I will add hitbox rotation next.

Reply:
Done!